oap-http/src/main/java/oap/http/HttpResponse.java

```diff
@@ -126,8 +129,56 @@ public HttpResponse withHeader( String name, String value ) {
         return this;
     }
 
+    public HttpResponse withCookie( String cookie ) {
+        if( StringUtils.isNotBlank( cookie ) ) {
+            headers.add( __( "Set-Cookie", cookie ) );
+        }
+        return this;
+    }
+
     public HttpResponse withContent( String content, ContentType contentType ) {
         this.contentEntity = new StringEntity( content( false, content, contentType ), contentType );
         return this;
     }
+
+    public static class CookieBuilder {
+        private static final Joiner JOINER = Joiner.on( ";" ).skipNulls();
+        private static final DateTimeFormatter FORMATTER =
+            DateTimeFormat.forPattern( "EEE, dd-MMM-yyyy HH:mm:ss zzz" );
+        private String SID;
+        private String domain;
+        private String expires;
+        private String path;
+        private String custom;
+
+        public CookieBuilder withSID( String SID ) {
+            this.SID = StringUtils.isNotBlank( SID ) ? "SID=" + SID : null;
+            return this;
+        }
+
+        public CookieBuilder withDomain( String domain ) {
+            this.domain = StringUtils.isNotBlank( domain ) ? "domain=" + domain : null;
+            return this;
+        }
+
+        public CookieBuilder withExpires( DateTime expires ) {
+            this.expires = expires != null ? "expires=" + FORMATTER.print( expires ) : null;
+            return this;
+        }
+
+        public CookieBuilder withPath( String path ) {
+            this.path = StringUtils.isNotBlank( path ) ? "path=" + path : null;
+            return this;
+        }
+
+        public CookieBuilder withCustomValue( String name, String value ) {
+            this.custom = StringUtils.isNoneBlank( name, value ) ? name + "=" + value : null;
```

Comment:
as a user I expect next code to work:
cb.withCustomValue("key1", "val1").withCustomValue("key2", "val2")
but 2nd invocation will override first one